src/app/shared/back/back.component.ts

```diff
@@ -0,0 +1,26 @@
+import { Component } from '@angular/core';
+import { Router, RoutesRecognized } from '@angular/router';
+import { filter, pairwise } from 'rxjs/operators';
+
+@Component({
+  selector: 'planet-back',
+  templateUrl: './back.component.html',
+  styleUrls: [ './back.component.scss' ]
+})
+export class BackComponent {
+
+  previousRoute: any;
+
+  constructor(private router: Router) {
+    this.router.events.pipe(
```

Comment:
@ArashDai Is there a reason you are subscribing to router events here?

There is a `Location` service built in to Angular with a `back()` method which should do the same.

Reply:
@paulbert I guess not, I didn't realize the Location service had a back method, I thought I had to go through the router to get to the last page.